### PR TITLE
Upstream merge 2022 08 29

### PR DESCRIPTION
**Pull request by torben-hansen**
# USE MERGE STRATEGY

Upstream Merge. See internal docs for details.

### Description of changes: 

A number of upstream BSSL commits.

### Testing:

CI

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by DominicDams — APPROVED** [review]

**Review by dkostic — APPROVED** [review]

**Review by darylmartin100 — DISMISSED** [review]

File: crypto/x509/algorithm.c [hunk 1]
```diff
@@ -122,7 +122,7 @@ int x509_digest_sign_algorithm(EVP_MD_CTX *ctx, X509_ALGOR *algor) {
 
 int x509_digest_verify_init(EVP_MD_CTX *ctx, const X509_ALGOR *sigalg,
                             EVP_PKEY *pkey) {
-  /* Convert the signature OID into digest and public key OIDs. */
+  // Convert the signature OID into digest and public key OIDs.
   int sigalg_nid = OBJ_obj2nid(sigalg->algorithm);
   int digest_nid, pkey_nid;
   if (!OBJ_find_sigid_algs(sigalg_nid, &digest_nid, &pkey_nid)) {
```
Comment (andrewhop) [inline crypto/x509/algorithm.c:128]: The comments down below got skipped in the conflict.

File: crypto/x509v3/v3_utl.c [hunk 2]
```diff
@@ -1273,56 +1237,55 @@ static int ipv6_from_asc(unsigned char v6[16], const char *in) {
...
 
-  /* Format result */
+  // Format result
 
   if (v6stat.zero_pos >= 0) {
-    /* Copy initial part */
+    // Copy initial part
     OPENSSL_memcpy(v6, v6stat.tmp, v6stat.zero_pos);
+
     /* Zero middle */
```
Comment (andrewhop) [inline crypto/x509v3/v3_utl.c:1289]:
```suggestion
    // Zero middle
```

**Review by andrewhop — DISMISSED** [review]

File: crypto/x509/algorithm.c (hunk 1 above)
Comment (torben-hansen) [reply crypto/x509/algorithm.c:128]: Sigh

File: crypto/x509v3/v3_utl.c (hunk 2 above)
Comment (torben-hansen) [reply crypto/x509v3/v3_utl.c]: Sigh^2...

**Review by andrewhop — APPROVED** [review]

**Pull request merged**